qutebrowser/browser/downloads.py

```diff
@@ -628,10 +639,7 @@ def get_request(self, request, page=None, fileobj=None, filename=None,
                          ui -> remove-finished-downloads is set to false.
 
         Return:
-            If the download could start immediately, (fileobj/filename given),
-            the created DownloadItem.
-
-            If not, None.
+            The created DownloadItem.
```

Comment:
It's important here the request is made (i.e. `self.fetch_request` is called) as late as possible, i.e. only *after* the user has answered the filename question.

If we open the request while waiting for the user, there's the risk the server will disconnect us for inactivity. For `WebPage.on_unsupported_content` (so for `DownloadManager.fetch`) we have no other way of handling this so we slowly download into RAM (see #79 and #270 for some background) - but if possible (like with `fetch`, when we have a `QNetworkRequest` rather than a `QNetworkReply` available) we should do the request when we're certain the download can actually be started.

With your change, `fetch` would be called immediately.